packages/components/pie-toggle-switch/src/index.ts

```diff
@@ -35,28 +38,35 @@ export class PieToggleSwitch extends RtlMixin(LitElement) implements ToggleSwitc
 
     render () {
         const {
+            aria,
             isChecked,
             isDisabled,
         } = this;
 
+        const toggleSwitchId = 'toggle-switch-description';
+
         return html`
             <label
                 data-test-id="toggle-switch-component"
                 class="c-toggleSwitch"
                 ?isChecked="${isChecked}"
-                ?isDisabled=${isDisabled}>
+                ?isDisabled=${isDisabled}
+                aria-label="${aria?.label || nothing}">
                 <input
+                    data-test-id="toggle-switch-input"
                     role="switch"
                     type="checkbox"
                     class="c-toggleSwitch-input"
                     .checked="${isChecked}"
                     .disabled="${isDisabled}"
-                    @change="${this.onToggleChange}">
+                    @change="${this.onToggleChange}"
+                    aria-describedby="${aria?.describedBy ? toggleSwitchId : ''}">
 
                 <div class="c-toggleSwitch-control">
                     ${isChecked ? html`<icon-check></icon-check>` : nothing}
                 </div>
             </label>
+            ${aria?.describedBy ? html`<div id="${toggleSwitchId}" data-test-id="${toggleSwitchId}" class="c-toggleSwitch-description">${aria?.describedBy}</div>` : nothing}
```

Comment:
How come we're using html rather than an attribute with the prop value?

Reply:
not sure why we might need this. the input element can accept `aria-describedby` that we can use. 
maybe it's worth checking how it's done in snacks? the component is called switch.

Reply:
Hey,

So I spoke to Jamie. `aria-describedby` takes an id for a description element. So what I have here is correct if we still want to allow an `aria.describedBy` prop

@kevinrodrigues I've added a screen recording to the pr description for adding the following as a prop. It reads the value, the label, the description and states that it is a switch.

```
  aria: {
    label: 'label',
    describedBy: 'description'
 }
 ```
 
 @raoufswe in snacks we just pass the id and the consumer, I'm guessing, needs to create their own description element.
 
 I've done similar to how we have in fozzie. Happy to change it to whatever you all prefer

Reply:
@FayeCarter is totally right - I completely mixed up `aria-describedby` with `aria-label` 🤦